nav2_map_server/README.md

```diff
@@ -109,36 +244,189 @@ and could be invoked by following command:
+> one needs to implement or use an existing I/O system.
+
+> Create a specialization for your map type e.g:
+```c++
+template<>class MapServer<new_mapT> : public nav2_util::LifecycleNode{};
+```
 
-`MapIO` library contains following API functions declared in `map_io.hpp` to work with
+>  if a new type of service/message is required one can write and directly add the service with a callback in the *specialization*.
+
+For more information on adding a new map-type, simply analyse 
+```c++ 
+// 3D map_server
+include/nav2_map_server/map_server_core.hpp
+include/nav2_map_server/map_3d/map_server_3d.hpp
+src/map_3d/map_server_3d.cpp
+include/nav2_map_server/map_server.hpp
+
+// 3D map_saver
+include/nav2_map_server/map_saver_core.hpp
+include/nav2_map_server/map_3d/map_saver_3d.hpp
+src/map_3d/map_saver_3d.cpp
+include/nav2_map_server/map_saver.hpp
+``` 
+
+## MapIO 2d library
+
+`MapIO 2D` library contains following API functions declared in `map_2d/map_io_2d.hpp` to work with
 OccupancyGrid maps:
 
-- loadMapYaml(): Load and parse the given YAML file
-- loadMapFromFile(): Load the image from map file and generate an OccupancyGrid
-- loadMapFromYaml(): Load the map YAML, image from map file and generate an OccupancyGrid
-- saveMapToFile(): Write OccupancyGrid map to file
+- map_2d::loadMapYaml(): Load and parse the given YAML file
+- map_2d::loadMapFromFile(): Load the image from map file and generate an `OccupancyGrid`
+- map_2d::loadMapFromYaml(): Load the map YAML, image from map file and generate an `OccupancyGrid`
+- map_2d::saveMapToFile(): Write `OccupancyGrid` map to file
+
+This one provides `OccupancyGrid` operations functionality to `Map Server/Saver` template specialization for 2D image like maps.
+ 
+## MapIO 3D library
+
+`MapIO 3D` library contains following API functions declared in `map_3d/map_io_3d.hpp` to work with
+`PointCloud` maps:
+
+- map_3d::loadMapYaml(): Load and parse the given YAML file
+- map_3d::loadMapFromFile(): Load the pcd from map file and generate an `PointCloud`
+- map_3d::loadMapFromYaml(): Load the map YAML, image from map file and generate an `PointCloud`
+- map_3d::saveMapToFile(): Write `PointCloud` map to file
+
+This one provides `PointCloud` handling functionality to `Map Server/Saver` template specialization for 3D point cloud maps.
 
 ## Services
 
-As in ROS navigation, the `map_server` node provides a "map" service to get the map. See the nav_msgs/srv/GetMap.srv file for details.
+As in ROS navigation, the map_server node provides a "map" service to get the map. See the nav_msgs/srv/GetMap.srv file for details.
 
-NEW in ROS2 Eloquent, `map_server` also now provides a "load_map" service and `map_saver` -
-a "save_map" service. See nav2_msgs/srv/LoadMap.srv and nav2_msgs/srv/SaveMap.srv for details.
+NEW in ROS2 Eloquent, map_server also now provides a "load_map" service and map_saver - a "save_map" service. See nav2_msgs/srv/LoadMap.srv and nav2_msgs/srv/SaveMap.srv for details.
```

Comment:
This change snippet is unnecessary. Restore back `map_server` code style and original formatting.